@using Services
@using Exceptions
@inject AuthService authService
@using Validation.Models
@using ImmagesApp.Shared.UI;
@inject NavigationManager navigationManager

<EditForm class="auth-form" Model="@LoginDataModel" OnValidSubmit="HandleValidSubmit">
    <DataAnnotationsValidator />

    <h2>Sing in</h2>
    <TextField Placeholder="Enter your email" @bind-Value="LoginDataModel.Email" ValidationFor="@(() => LoginDataModel.Email)" />
    <TextField Type="password" Placeholder="Enter your password" @bind-Value="LoginDataModel.Password" ValidationFor="@(() => LoginDataModel.Password)" />

    <input type="submit" class="btn btn-primary" value="Save"/>
</EditForm>

@code {
    [CascadingParameter]
    public ErrorLogger Logger { get; set; } = null!;
    
    
    public LoginDataModel LoginDataModel = new();

    private async Task HandleValidSubmit() {
        try
        {
            await authService.Login(LoginDataModel.Email!, LoginDataModel.Password!);
            navigationManager.NavigateTo("/", true);
        }
        catch (ServiceException error)
        {
            Logger.Log(error);
        }
    }
}
